src/components/cc-product-list/cc-product-list.js

```diff
@@ -0,0 +1,235 @@
+import { css, html, LitElement } from 'lit';
+import { i18n } from '../../lib/i18n.js';
+import { accessibilityStyles } from '../../styles/accessibility.js';
+import { linkStyles } from '../../templates/cc-link/cc-link.js';
+import '../cc-product-card/cc-product-card.js';
+import '../cc-badge/cc-badge.js';
+import '../cc-icon/cc-icon.js';
+import '../cc-input-text/cc-input-text.js';
+import { ProductsController } from './products-controller.js';
+
+/**
+ * @typedef {import('./cc-product-list.types.js').ProductsCategory} ProductsCategory
+ * @typedef {import('lit').PropertyValues<CcProductList>} CcProductListPropertyValues
+ * @typedef {import('lit').TemplateResult<1>} TemplateResult
+ */
+
+/**
+ * A component that displays a list of products in sections and allows to search and filter through it.
+ *
+ * @cssdisplay block
+ */
+export class CcProductList extends LitElement {
+
+  static get properties () {
+    return {
+      filterCategory: { type: String, attribute: 'filter-category' },
+      productsByCategories: { type: Array, attribute: 'products-by-categories' },
+      searchInput: { type: String, attribute: 'search-input' },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {ProductsCategory[]} the list of products in their respective categories.  */
+    this.productsByCategories = [];
+
+    /** @type {string} a string to presearch through the list */
+    this.searchInput = '';
+
+    /** @type {string|null} a string to prefilter by a given category */
+    this.filterCategory = null;
+
+    /** @type {ProductsController} */
+    this._productsCrtl = new ProductsController(this);
+  }
+
+  /**
+   * @param {Event & { target: HTMLInputElement }} e
+   */
+  _onCategoryChange (e) {
+    this.filterCategory = e.target.value;
+  }
+
+  /**
+   * @param {Object} event
+   * @param {string} event.detail
+  */
+  _onSearchInput ({ detail: value }) {
+    this.searchInput = value;
+  }
+
+  /**
+   * @param {CcProductListPropertyValues} changedProperties
+  */
+  willUpdate (changedProperties) {
+
+    if (changedProperties.has('productsByCategories')) {
+      this._productsCrtl.productsByCategories = this.productsByCategories;
+    }
+
+    if (changedProperties.has('filterCategory')) {
+      this._productsCrtl.toggleCategory(this.filterCategory);
+      this._productsCrtl.search(this.searchInput);
+    }
```

Comment:
issue: I was expecting the `this._productsCrtl` to remember the last `searchInput`.

This logic should be inside the ProductController. Whether you do search then toggle category or the other way around should work in the controller.